how-it-works/1_protocol/03_message-routing.subpage.md

```diff
@@ -7,8 +7,128 @@ slug: message-routing
+* Xnet messages to be sent to other subnets, and
+* Canister metadata (module hashes, *certified variables*).
+
+The responses to ingress messages are referred to as ingress history.
+The certified responses can be read and validated against the subnet's public key by users as the response to their ingress messages.
+This way of validating responses to state-changing messages to a blockchain is extremely powerful when compared to other approaches seen in the field like reading the response from a transaction log.
+
+This per-round state certification ensures that any item of data relevant for interactions of users and subnets and between different subnets on the Internet Computer is authenticated.
+This particularly enables secure and verifiable inter-subnet communication, a crucial feature of the Internet Computer as well as an enabler of its scalability.
+
+### Per-Checkpoint Certification
+
+Wasm code changed through canister updates and written-to ("dirty") memory pages of canisters are not certified in every round, but flagged as modified by the execution layer whenever they are written, using approaches known from operating system kernels.
+At every checkpointing interval, that is, every multiple hundred rounds, which corresponds to around 10 minutes, the subnet computes a so-called checkpoint at which the state of a subnet is certified.
+The state certification is done incrementally by incorporating the content of all pages changed since the last checkpoint certification into the Merkle state tree and propagating the changes up the tree.
+Finally, the root of the Merkle tree and additional metadata is signed with a BLS threshold signature by the subnet, thereby certifying every leaf of the Merkle tree.
+The signed result is called a *catch-up package* as it can be used by nodes to efficiently catch up to the point in time when the checkpoint was made.
+Technically, the overall state is partitioned into chunks and the catch-up package contained an authenticated hash of each of the chunks.
+The run time of this checkpointing operation is linear in the number of changed pages and not the overall state size on the subnet.
+This is crucial as a subnet can hold terabytes of state in the future and a full recertification of multiple terabytes of replicated state would not be practical at every checkpoint interval.
+
+## State Synchronization
+
+The message routing layer implements another feature quite unique to the Internet Computer.
+As described above, every checkpointing interval, i.e., around every 10 minutes, the complete subnet state is certified by the subnet through a BLS threshold signature on a Merkle tree and made available as a catch-up package.
+One prominent use case of a catch-up package is to allow new nodes to join a subnet, e.g., if the subnet is to grow in size or a node needs to be replaced because of having been destroyed in a disaster, or if a node has to catch-up because it was down for some time.
```

Comment:
I'd switch this around to mention the use case the name already suggests first:
```suggestion
As the name already suggests, a catch-up package allows a node to catch-up if it has fallen behind, e.g., because it was down for some time. In addition, it allows new nodes to join, e.g., if the subnet is to grow in size or a node needs to be replaced because of having been destroyed in a disaster.
```